sonar-orchestrator/src/main/java/com/sonar/orchestrator/container/Server.java

```diff
@@ -204,18 +227,86 @@ public SonarClient adminWsClient() {
    * if it does not exist yet, otherwise it is reset.
    */
   public Server restoreProfile(Location backup) {
+    Version version = version();
     try (InputStream input = fileSystem.openInputStream(backup)) {
-      newHttpCall("/api/qualityprofiles/restore")
-        .setMethod(HttpMethod.MULTIPART_POST)
-        .setAdminCredentials()
-        .setParam("backup", IOUtils.toString(input, UTF_8))
-        .execute();
+      if (version.isGreaterThanOrEquals("6.0")) {
+        newHttpCall("/api/qualityprofiles/restore")
+          .setMethod(HttpMethod.MULTIPART_POST)
+          .setAdminCredentials()
+          .setParam("backup", IOUtils.toString(input, UTF_8))
+          .execute();
+      } else {
+        restoreProfileForOldVersion(new ByteArrayBody(IOUtils.toByteArray(input), ContentType.TEXT_XML, "profile-backup.xml"));
+      }
       return this;
     } catch (IOException e) {
       throw new IllegalStateException("Cannot restore Quality profile", e);
     }
   }
 
+  private Server restoreProfileForOldVersion(ContentBody backup) {
+    if (url == null) {
+      throw new IllegalStateException("Can not restore profiles backup if the server is not started");
+    }
+    // still no sonar-ws-client for this web service
+    DefaultHttpClient client = new DefaultHttpClient();
+    try {
+      HttpHost targetHost = new HttpHost("localhost", new URL(getUrl()).getPort(), "http");
+
+      HttpParams params = client.getParams();
+      HttpConnectionParams.setConnectionTimeout(params, 60_000);
+      HttpConnectionParams.setSoTimeout(params, 120_000);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "HttpConnectionParams"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)